core/native/src/Instant.kt

```diff
@@ -23,54 +23,6 @@ public actual enum class DayOfWeek {
     SUNDAY;
 }
 
-// This is a function and not a value due to https://github.com/Kotlin/kotlinx-datetime/issues/5
-// org.threeten.bp.format.DateTimeFormatterBuilder.InstantPrinterParser#parse
-private val instantParser: Parser<Instant>
-    get() = localDateParser
-        .chainIgnoring(concreteCharParser('T').or(concreteCharParser('t')))
-        .chain(intParser(2, 2)) // hour
-        .chainIgnoring(concreteCharParser(':'))
-        .chain(intParser(2, 2)) // minute
-        .chainIgnoring(concreteCharParser(':'))
-        .chain(intParser(2, 2)) // second
-        .chain(optional(
-            concreteCharParser('.')
-                .chainSkipping(fractionParser(0, 9, 9)) // nanos
-        ))
-        .chainIgnoring(concreteCharParser('Z').or(concreteCharParser('z')))
```

Comment:
If the other parsers are implemented via `ZonedDateTime`, there is no need to have the parser for `Instant` in the common code, and the only place that requires an explicit implementation is the native part. Then, it's probably easier to modify this parser at this line so that it accepts any `ZoneOffset` and not just `Z` (and propagate this change correspondingly).